code/online_challenges/src/project_euler/problem_034/problem_034.cpp

```diff
@@ -0,0 +1,27 @@
+#include <iostream>
+#include <vector>
+int factorial (int n) 
+{
+    if(n == 0)
```

Comment:
TBH, I would avoid the recursive definition, as recursion in C++ takes up a lot of stack space. I would just use a loop. That would remove the need for the if statement as well. If you want to shorten this and keep this implementation, then you can do the following:

```C++
return (n <= 1) ? (1) : (n * factorial(n - 1));
```